Bug#1520090 - Updated AWS Configuration to include cluster labeling

gaurav-nelson opened the pull request:
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1520090
 - Removed the file `admin_guide/aws_cluster_labeling.adoc`
 - Added the contents of deleted files in `install_config/configuring_aws.adoc`
 - Updated `_topic_map.yml` (removed entry for deleted file)
 - Checked for existing links to the deleted file and updated the single link found in 3.7 release notes.

gnunn1 commented:
Can ```openshift_clusterid``` be documented here as well?

gaurav-nelson commented:
@gnunn1 sure, I'll add that.

gaurav-nelson commented on install_config/configuring_aws.adoc:
```diff
@@ -166,6 +166,66 @@ AWS_SECRET_ACCESS_KEY=<secret_key>
+`kubernetes.io/cluster/<name>` tag avoids potential issues with multiple zones
+or multiple clusters.
+
+[NOTE]
```
@gnunn1 I have added this note which includes `clusterid` info. Does this looks alright?

gaurav-nelson replied:
I am also trying to get more info about `openshift_clusterid` variable meanwhile.

gaurav-nelson commented:
@gnunn1 Can you please check the updates? https://github.com/gaurav-nelson/openshift-docs/blob/04b266567aa0842b94cace1b57702fc5520273a9/install_config/configuring_aws.adoc#aws-cluster-labeling

gnunn1 commented:
Looks good but you are still meaning documenting the ```openshift_clusterid``` variable in the ```Configuring {product-title} for AWS with Ansible```. This variable is mentioned in the 3.7 release notes under the section ```Labeling Clusters for Amazon Web Services```:

```
Labeling Clusters for Amazon Web Services
Starting with 3.7 versions of the installer, if you configured AWS provider credentials, you must also ensure that all instances are labeled. Then, set the openshift_clusterid variable to the cluster ID. See Labeling Clusters for Amazon Web Services (AWS) for more information.
```

https://docs.openshift.com/container-platform/3.7/release_notes/ocp_3_7_release_notes.html#ocp-37-notable-technical-changes

gaurav-nelson commented:
@gnunn1 PTAL
I have:
- updated the Configuring AWS section to include **Setting Ansible Variables** section which talks about how to set the Ansible variable for `openshift_clusterid`. (Preview: https://github.com/gaurav-nelson/openshift-docs/blob/2963a728280ef7d5a74be47e42f99c2b87c75cd7/install_config/configuring_aws.adoc#aws-setting-ansible-variables) 
- include `openshift-clusterid` variable to the Cluster Variable section (last entry of the table, preview: https://github.com/gaurav-nelson/openshift-docs/blob/2963a728280ef7d5a74be47e42f99c2b87c75cd7/install_config/install/advanced_install.adoc#configuring-cluster-variables)

gaurav-nelson commented:
@gnunn1 bump!

gnunn1 commented:
@gaurav-nelson I don't believe that the variable ```openshift_clusterid``` should be in the ```Setting Ansible Variables```, it should be in the section ```Configuring {product-title} for AWS with Ansible``` with the other variables. Was ```Setting Ansible Variables``` always there, if not just remove it and put ```openshift_clusterid``` in the other section with the other ansible variables.

Otherwise it looks fine to me.

gaurav-nelson commented:
Thanks @gnunn1 
Latest commit fixes these nits:

- Removed the `Setting Ansible Variables` section
- Added `openshift_clusterid` variable to `Configuring {product-title} for AWS with Ansible` section

gaurav-nelson commented:
@gnunn1 PTAL

gnunn1 commented:
Just add ```openshift_clusterid``` in the example with the other variables and I am fine.

gaurav-nelson commented:
Thanks @gnunn1 I have included it in the list with other variables. I

gaurav-nelson commented:
@openshift/team-documentation PTAL

bfallonf commented on install_config/configuring_aws.adoc:
```diff
@@ -218,3 +220,61 @@ $ systemctl restart origin-node
+
+[[aws-cluster-labeling]]
+== Labeling Clusters for AWS
+Starting with {product-title} version 3.7 of the installation program, if you
```
What does "for the installation program" refer to? I don't think this is an OpenShift thing. Would it be easier to take it out?

bfallonf commented on install_config/configuring_aws.adoc:
```diff
@@ -218,3 +220,61 @@ $ systemctl restart origin-node
+Services] to learn more about labeling and tagging in {product-title}.
+
+[[aws-resources-that-need-tags]]
+=== Resources That Need Tags
```
I can see this section was like this already, but i'm wondering if it can be merged into the 'Tagging an Existing Cluster' section below. This isn't adding much that the section below isn't already saying. WDYT?

gaurav-nelson replied:
Thanks @bfallonf I am keeping this as it is for now because I think the next section is a procedure and this section is just calling what needs to be tagged. You are right that both of these sections should be combined, I will revisit this later.

bfallonf commented:
@gaurav-nelson A few comments, but looks good overall.

gaurav-nelson commented:
moved existing content around, no revision history needed.
